ring/ring.go

```diff
@@ -307,6 +307,11 @@ func (r *Ring) updateRingState(ringDesc *Desc) {
 		}
 	}
 
+	// Ensure the ID of each InstanceDesc is set based on the map of instances. This
+	// handles the case where some components are running older lifecyclers which do
+	// not set the instance ID when registering in the ring.
+	ringDesc.setInstanceIDs()
```

Comment:
This requires updated dskit on *writer* side, instead of *reader* side. This can confuse readers expecting instance ID already.

Reply:
I think that's backwards. This struct (`Ring`) is what is in-memory in distributors for the ingester ring. The previous iteration (https://github.com/grafana/dskit/pull/387/commits/821bbdd99927ea0727682a5144d16399d93d2ab8) was wrong and did require updated code on the writer side but this version is the reader side unless I'm missing something.

Reply:
You're correct. I've got confused by commit message ("Set instance IDs when updating Ring, not when loading from KV store") and then haven't checked the actual code change. My bad.